src/my-bot.ts

```diff
@@ -1,38 +1,132 @@
+        var upPos = pos.getUpPosition();
+        var upString = (upPos) ? (upPos.getString()) : "";
+        var downPos = pos.getDownPosition();
+        var downString = (downPos) ? (downPos.getString()) : "";
+        var leftPos = pos.getLeftPosition();
+        var leftString = (leftPos) ? (leftPos.getString()) : "";
+        var rightPos = pos.getRightPosition();
+        var rightString = ((rightPos) ? (rightPos.getString()) : "");
+        var thisString = pos.getString();
+        if(hitMap[thisString] && hitMap[upString]) {
+            triedMap = this.markLeftRight(pos, triedMap);
+            triedMap = this.markLeftRight(upPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(upPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[downString]) {
+            triedMap = this.markLeftRight(pos, triedMap);
+            triedMap = this.markLeftRight(downPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(downPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[leftString]) {
+            triedMap = this.markUpDown(pos, triedMap);
+            triedMap = this.markUpDown(leftPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(leftPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[rightString]) {
+            triedMap = this.markUpDown(pos, triedMap);
+            triedMap = this.markUpDown(rightPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(rightPos, triedMap);
         }
-        return String.fromCharCode(newRow);
+        return triedMap;
     }
 
-    private getNextColumn(column) {
-        return column % 10 + 1;
+    private markLeftRight(pos, triedMap: { [ pos: string] : boolean }) : { [ pos: string] : boolean }{
+        var leftPos = pos.getLeftPosition();
+        var leftString = (leftPos) ? (leftPos.getString()) : "";
+        var rightPos = pos.getRightPosition();
+        var rightString = ((rightPos) ? (rightPos.getString()) : "");
+        if(leftString && leftString != "") {
+            triedMap[leftString] = true;
+        }
+        if(rightString && rightString != "") {
+            triedMap[rightString] = true;
+        }
+        return triedMap;
+    }
+
+    private markUpDown(pos, triedMap: { [ pos: string] : boolean }): { [ pos: string] : boolean }{
+        var upPos = pos.getUpPosition();
+        var upString = (upPos) ? (upPos.getString()) : "";
+        var downPos = pos.getDownPosition();
+        var downString = (downPos) ? (downPos.getString()) : "";
+        if(upString && upString != "") {
+            triedMap[upString] = true;
+        }
+        if(downString && downString != "") {
+            triedMap[downString] = true;
+        }
+        return triedMap;
+    }

```

Comment:
I do, however, really like that you've cut all these marking methods out into their own little sections - this is already massively reducing duplication - I just think you can do even better :P